shared/filestore/s3store.go

```diff
@@ -41,6 +41,7 @@ const (
 	bucketNotFound = "NoSuchBucket"
 )
 
+var ErrNoS3Bucket = errors.New("no such bucket")
```

Comment:
Nit: 
Just add a small comment explaining it. Ex.
```suggestion
// ErrNoS3Bucket is returned when testing connection and no S3 bucket is found (feel free to word it better)
var ErrNoS3Bucket = errors.New("no such bucket")
```